src/jsbsim_bridge.cpp

```diff
@@ -94,28 +95,60 @@ JSBSimBridge::JSBSimBridge(JSBSim::FGFDMExec *fdmexec, std::string &path)
 
 JSBSimBridge::~JSBSimBridge() {}
 
-bool JSBSimBridge::CheckConfigElement(TiXmlHandle &config, std::string group, std::string name) {
-  TiXmlElement *group_element = config.FirstChild(group).Element();
-  if (!group_element) {
-    return false;
+bool JSBSimBridge::SetFdmConfigs(ConfigurationParser *cfg) {
+  const TiXmlElement *config = cfg->LoadXmlHandle()->FirstChild("jsbsimbridge").Element();
+
+  _fdmexec->SetRootDir(SGPath(JSBSIM_ROOT_DIR));
+
+  std::string aircraft_path;
+  if (config && CheckConfigElement(*config, "aircraft_directory")) {
+    GetConfigElement<std::string>(*config, "aircraft_directory", aircraft_path);
+  } else {
+    aircraft_path = "models/" + cfg->getModelName();
+  }
+  _fdmexec->SetAircraftPath(SGPath(aircraft_path.c_str()));
+  _fdmexec->SetEnginePath(SGPath("Engines"));
+
+  if (!cfg->isHeadless()) {  // Check if HEADLESS mode is enabled
+    _fdmexec->SetOutputDirectives(SGPath("data_out/flightgear.xml"));
   }
 
-  TiXmlElement *e = group_element->FirstChildElement(name);
-  return e != nullptr;
+  std::string aircraft_model;
+  if (config && CheckConfigElement(*config, "aircraft_model")) {
+    GetConfigElement<std::string>(*config, "aircraft_model", aircraft_model);
+  } else {
+    aircraft_model = cfg->getModelName();
+  }
+  _fdmexec->LoadModel(aircraft_model.c_str(), false);
+
+  // Load Initial Conditions
+  JSBSim::FGInitialCondition *initial_condition = _fdmexec->GetIC();
+  SGPath init_script_path = SGPath::fromLocal8Bit((cfg->getInitScriptPath()).c_str());
+  initial_condition->Load(SGPath(init_script_path), false);
+
+  return true;
 }
 
 bool JSBSimBridge::SetMavlinkInterfaceConfigs(std::unique_ptr<MavlinkInterface> &interface, TiXmlHandle &config) {
   TiXmlElement *mavlink_configs = config.FirstChild("mavlink_interface").Element();
 
   if (!mavlink_configs) return true;  // Nothing to set
 
-  if (mavlink_configs->FirstChildElement("tcp_port")) {
-    interface->SetMavlinkTcpPort(std::stoi(mavlink_configs->FirstChildElement("tcp_port")->GetText()));
+  int tcp_port;
+  if (CheckConfigElement(config, "mavlink_interface", "tcp_port")) {
+    GetConfigElement<int>(config, "mavlink_interface", "tcp_port", tcp_port);
```

Comment:
Make sure to avoid such duplication, so that the call can look like this:
```suggestion
  if (GetConfigElement<int>(config, "mavlink_interface", "tcp_port", tcp_port)) {
```